### PR TITLE
Ensure config cache is not stored for gradle init

### PR DESCRIPTION
**Pull request by abstratt**
in integration test for `gradle init` vs CC.

Issue: #20156

### CODE REVIEW

**Comment by abstratt** [comment]
@bot-gradle test RFR

File: subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheIntegrationTest.groovy [hunk 1]
```diff
@@ -425,5 +432,8 @@ class ConfigurationCacheIntegrationTest extends AbstractConfigurationCacheIntegr
         then:
         outputContains("> Task :init")
         succeeds 'properties'
+        def projectName2 = testDirectory.name
```
[the remark this answers is not in the document]
Comment (abstratt) [reply subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheIntegrationTest.groovy:435]: Nope, the bug happens regardless of the current dir.

File: subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheIntegrationTest.groovy [hunk 2]
```diff
@@ -410,13 +411,19 @@ class ConfigurationCacheIntegrationTest extends AbstractConfigurationCacheIntegr
     }
 
     def "can init two projects in a row"() {
+        ConfigurationCacheFixture fixture = new ConfigurationCacheFixture(this)
```
[the remark this answers is not in the document]
Comment (abstratt) [reply subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheIntegrationTest.groovy:414]: can do, but copied from existing code

[the remark this answers is not in the document]
Comment (abstratt) [reply subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheIntegrationTest.groovy:414]: actually, it is consistent with the many cases in this class

[the remark this answers is not in the document]
Comment (abstratt) [reply subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheIntegrationTest.groovy]: Done

File: subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheIntegrationTest.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (abstratt) [reply subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheIntegrationTest.groovy:435]: (and that is why I figured it was really important to fix)

**Review by bamboo — APPROVED** [review]

**Comment by abstratt** [comment]
@bot-gradle test and merge

**Pull request merged**